lldb/source/Symbol/SwiftASTContext.cpp

```diff
@@ -1448,25 +1448,45 @@ void ApplyWorkingDir(SmallString &clang_argument, StringRef cur_working_dir) {
   llvm::sys::path::append(clang_argument, cur_working_dir, rel_path);
   llvm::sys::path::remove_dots(clang_argument);
 }
+
+enum ClangArgCategory {
+  ClangArgCategoryNone = 0,
+  ClangArgCategoryUnique = 1 << 0,
+  ClangArgCategoryMacro = 1 << 1,
+};
+
+inline bool IsMultiArgClangFlag(const StringRef& arg) {
+  return arg == "-D" || arg == "-U" || arg == "-working-directory";
```

Comment:
Previously we only had to check for multi-arg flags in the 3 cases we actually were doing something special to it. But with uniquing, do we need to check for more cases?
I guess we are still only uniquing the arguments which we recognize in `GetClangArgCategory`, right?

Reply:
For now that's true (except that it would be `(GetClangArgCategory(arg) & ClangArgCategoryUnique) || arg == "-working-directory` as we use `-working-directory` only to apply to paths to make them absolute). In future there'll be at least `-fmodule-map-file=` which is single-arg by design, although it wouldn't break this method logic anyway. I could change this code to use output of `GetClangArgCategory` if you're okay with single-arg case.

Reply:
Actually, I've forgot to mention that there're different conditions: `==` vs `startswith`. Would you advise to separate some list of constants to unify lists of args to join/unique? I think of something like this:

```
const StringRef kMacroFlags[] = { "-D", "-U" };
const StringRef kUniqueMultiArgFlags[] = { "-I", "-F" }; // this would appear in the next PR, need an elegant way to append kMacroFlags here (using macro? :) )
```

And then just use these arrays to check for `==` or `startswith` where needed

Reply:
@adrian-prantl what do you think about this?

Reply:
Sure, that seems reasonable. The naming should follow the LLDB style though:
`std::array<StringRef> macro_flags = {...};`
`std::array<StringRef> contractable_multiword_flags = {..};`

Reply:
Ok, will add macro_flags then